### ISSUE TITLE
azure-sb notificationHubservice.send method using TCP protocol which will causing function app SNAT exhaustion issue when request increase

### ISSUE THREAD

**Issue by yuu045-ran** [Azure/azure-sdk-for-node#5207]
Hello experts,

**Issue description**

azure-sb notificationHubservice.send method using TCP protocol which will causing function app SNAT exhaustion issue when request count increase, based on network packet capture result, each .send seems to occupy one SNAT port when using function app, this will cause outbound connection issue.(https://devblogs.microsoft.com/premier-developer/reducing-snat-port-consumption-in-azure-app-services/)

**Analysis**

https://github.com/Azure/azure-sdk-for-node/blob/fa35b874c212ed71a56f8093ca8e1d959f661cfb/examples/ASM/notification-hubs.md

Starting to track the .send method from here:
https://github.com/Azure/azure-sdk-for-node/blob/fa35b874c212ed71a56f8093ca8e1d959f661cfb/lib/services/serviceBus/lib/notificationhubservice.js#L136

Eventually, the TCP connection of .send method seems to be by-design because request method is finally used, and I'm not sure if there is a way to force the HTTP client to be re-used or keep alive - 
https://github.com/Azure/azure-sdk-for-node/blob/fa35b874c212ed71a56f8093ca8e1d959f661cfb/lib/common/lib/services/serviceclient.js#L235

**Questions**

- Is the above sympton a known issue for notificationHubservice.send method ? Will we have an ETA on the fix?

- Do we have any known-workarounds to force the TCP connection reuses SNAT port(e.g. similiar to https://www.npmjs.com/package/keep-alive-agent) in code?
- If the issue needs time to be fixed and it's not possible to modify parameters in notificationHubservice.send method, do we have any known workarounds(e.g. using API then reuse API's HTTP client)?

Thank you in advance.

**Comment by ramya-rao-a** [comment]
@mpodwysocki Can you help here?

**Comment by mpodwysocki** [comment]
@yuu045-ran Yes, this is indeed an issue, and although in Node 0.12, connection pooling was added, we are not taking advantage of this.  To enable this, we would need to do the following to use the agent options to use the `http.Agent` class.

```js
request({
    method: method,
    uri: uri,
    agentOptions: {keepAlive:true, maxSockets:1},
    agentClass: http.Agent
})
```

**Comment by yuu045-ran** [comment]
Hello @mpodwysocki , thank you very much for your reply. Will we have any ETA or trackable link of this bug? I'm actually from CSS team, my customer is having ongoing potential to get SNAT port exhausion and seems that we do not have a workaround here.

**Comment by mpodwysocki** [comment]
This issue should be able to be closed since merging #5209

**Comment by dw511214992** [comment]
the package is released.
